instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/InjectionObject.java

```diff
@@ -0,0 +1,73 @@
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet;
+
+import io.opentelemetry.javaagent.bootstrap.servlet.SnippetHolder;
+import java.io.IOException;
+import javax.servlet.ServletOutputStream;
+
+public class InjectionObject {
+  public int headTagBytesSeen = -1;
+  public String characterEncoding;
+  public SnippetInjectingResponseWrapper wrapper;
+
+  public boolean inject() {
+    return headTagBytesSeen == 5;
+  }
+
+  public boolean injected() {
```

Comment:
and move calls inside of `*Injection`
```suggestion
  private boolean injected() {
```